R/read.any.R

```diff
@@ -204,34 +207,68 @@ read.any <- function(
   }
 
   if(!is.null(widths) && is.null(col.names) || tolower(file_type) == 'fwf'){
-    x = read.fwf(filename, widths, header = FALSE)
+    x = utils::read.fwf(filename, widths, header = FALSE)
   }
 
   if(!is.null(widths) && !is.null(col.names) || tolower(file_type) == 'fwf'){
-    x = read.fwf(filename, widths, header = FALSE, col.names = col.names)
-  }  
- 
+    x = utils::read.fwf(filename, widths, header = FALSE, col.names = col.names)
+  }
+    
+  if(grepl("\\.pbix$", filename)) {
+      
+    # Validate pbi-tools is downloaded.
+    if (shell('pbi-tools info') != 0) {
+      message("pbi-tools is not configured correctly")
+    }
+      
+    # Extract tables with export-data into a temporary directory.
+    tmpDir = "temp_subdirectory"
```

Comment:
please use `tmpDir = glue::glue('{tempdir()}/pbi-extract')`. this will keep it in the computers temp directory to reduce risk of accidentally deleting important data with recursive unlink (a very dangerous operation) and make it easier to clean up temp files if anything accidentally gets left over. 

with this change, you won't need to have different code based on `folder` so you can clean up the lines below.